lib/facter/tpm2/util.rb

```diff
@@ -21,14 +21,21 @@ module Facter::TPM2; end
 #
 class Facter::TPM2::Util
   def initialize
-    @prefix = Facter::TPM2::Util.tpm2_tools_prefix
-  end
-
-  # Facter executes a CLI command using the tpm2-tools path
-  # @param [String] cmd The CLI command string for Facter to execute
-  def exec(cmd)
-    Facter.debug "executing '#{File.join(@prefix, cmd)}'"
-    Facter::Core::Execution.execute(File.join(@prefix, cmd))
+    cmd  = Facter::Core::Execution.which('tpm2_getcap')
```

Comment:
what if cmd returns nil?

Reply:
It would never be called because of the `confine` in the top-level fact.